src/Assets/Asset.php

```diff
@@ -601,14 +604,30 @@ public function saveQuietly()
      */
     public function save()
     {
+        $isNew = is_null(Facades\Asset::find($this->id()));
```

Comment:
When I upload an asset via the Control Panel, the `AssetCreating` and `AssetCreated` events don't seem to get dispatched.

Reply:
Maybe thats because it uploads the file, at which point it gets added to the asset container contents so 'exists' as an asset.

Reply:
Actually no, I bets it this... https://github.com/statamic/cms/blob/b18025a53642a103bd0a642232a3a00f5cedf35f/src/Assets/Asset.php#L870

Anyone know why thats there? syncOriginal gets called during save() anyway.

Reply:
I don't love the code repetition but https://github.com/statamic/cms/pull/9378/commits/3ca2a98880de2d22b6fc25224ef3bd3227834eea would solve it